gwsumm/archive.py

```diff
@@ -178,8 +178,24 @@ def read_data_archive(sourcefile):
     ----------
     sourcefile : `str`
         path to source HDF5 file
+
+    rm_source_on_fail : `bool`, optional
+        remove the source HDF5 file if there was an OSError when opening the
+        file
     """
-    from h5py import File
+
+    # Check that the HDF5 source file is able to be opened.
+    # A common failure mode is that the HDF5 file is corrupted and this brings
+    # down the whole workflow, requiring manual intervention. Here, we attempt
+    # to automatically catch a common failure
+    try:
+        h5file = File(sourcefile, 'r')
+    except OSError as exc:  # file is corrupt, so we remove it to start fresh
+        if not rm_source_on_fail:
+            raise
+        warnings.warn(f"failed to read {sourcefile} [{exc}], removing...")
+        os.remove(sourcefile)
+        return
 
     with File(sourcefile, 'r') as h5file:
```

Comment:
With the new code above you already have `File(...)` so this should just be something like:

```suggestion
    with h5file:
```